### PR TITLE
[merged] Exit with proper exit code

### PR DESCRIPTION
**Pull request by rhatdan**

### CODE REVIEW

**Comment by rhatdan** [comment]
Fixes https://github.com/projectatomic/atomic/issues/386

**Comment by rhatdan** [comment]
@baude PTAL

**Comment by baude** [comment]
You will likely want something like this to protect from tuple errors when the exit is legit.  This is why the tests are failing.

+        rc = 0 if len(e.args) < 1 else e.args[0]
+        sys.exit(rc)

File: atomic [hunk 1]
```diff
@@ -613,10 +613,11 @@ if __name__ == '__main__':
             need_root()
     except Exception as e:
         sys.stderr.write("%sn" % str(e))
-    except SystemExit:
+    except SystemExit as e:
         # Overriding debug args to avoid a traceback from sys.exit()
         if 'args' in locals():
             args.debug = False
+        sys.exit(e.args[0])
```
Comment (cgwalters) [inline atomic]: Confused by `e.args[0]` - isn't that `e.code`?

**Comment by rhatdan** [comment]
@baude I went with @cgwalters fix, does this fix the problem you were seeing?

**Comment by baude** [comment]
yes'sir.

**Comment by rhatdan** [comment]
@rh-atomic-bot r+

**Pull request closed without merging**